src/routes/api/tasks/index.ts

```diff
@@ -151,32 +187,122 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
           404: Type.Object({ message: Type.String() })
         },
         tags: ['Tasks']
-      }
+      },
+      preHandler: (request, reply) => request.isModerator(reply)
     },
     async function (request, reply) {
       const { id } = request.params
       const { userId } = request.body
 
-      const task = await fastify.repository.find<Task>('tasks', { where: { id } })
+      const task = await fastify.knex<Task>('tasks').where({ id }).first()
       if (!task) {
-        return notFound(reply)
+        return reply.notFound('Task not found')
       }
 
-      await fastify.repository.update('tasks', {
-        data: { assigned_user_id: userId },
-        where: { id }
-      })
+      await fastify
+        .knex('tasks')
+        .where({ id })
+        .update({ assigned_user_id: userId ?? null })
 
       task.assigned_user_id = userId
 
       return task
     }
   )
-}
 
-function notFound (reply: FastifyReply) {
-  reply.code(404)
-  return { message: 'Task not found' }
+  fastify.post(
+    '/:id/upload',
+    {
+      schema: {
+        params: Type.Object({
+          id: Type.Number()
+        }),
+        consumes: ['multipart/form-data'],
+        response: {
+          200: Type.Object({
+            path: Type.String(),
+            message: Type.String()
+          }),
+          404: Type.Object({ message: Type.String() }),
+          400: Type.Object({ message: Type.String() })
+        },
+        tags: ['Tasks']
+      }
+    },
+    async function (request, reply) {
+      const { id } = request.params
+
+      return fastify.knex.transaction(async (trx) => {
+        const task = await trx<Task>('tasks').where({ id }).first()
+        if (!task) {
+          return reply.notFound('Task not found')
+        }
+
+        const file = await request.file()
+
+        if (!file) {
+          return reply.notFound('File not found')
+        }
+
+        const allowedMimeTypes = ['image/jpeg', 'image/png']
+        if (!allowedMimeTypes.includes(file.mimetype)) {
```

Comment:
I realize mime type is not guessed from file content, if I send `one_line_csv.png`, it does work...

Reply:
Should I use `file-type`?
```ts
const fileBuffer = await file.toBuffer();
const fileType = await fileTypeFromBuffer(fileBuffer);
```

Reply:
No opinion about this @climba03003 @gurgunday?

Reply:
After this feedback, I will merge.

Reply:
I'd say yes, as it's a security risk. It's something I checked in one of my APIs: https://github.com/Fdawgs/docsmith/blob/main/src/routes/pdf/html/index.js#L32-L51

Reply:
Great the link, this is a good reason for educating about `addContentTypeParser` utility.
But don't you think we should mention this security risk in the plugin documentation?

Reply:
There could be a mention, yeah

But it's ultimately the same idea of never trusting a client without input validation

Reply:
But do we offer this kind of validation, or does a package we use offer this kind of validation?

Doesn't look like we recommend real validation for content file by reading the doc.

Reply:
I am not very strong opinion on inspecting the file content for the content-type, most of the application only use `.ext` to `content-type` (e.g. Reverse Proxy, Browser, OS).
If it is an input file (batch processing), you should validate the content inside instead of the guessed `content-type`.
If it is an image file or the others, using wrong content-type does make changes (malicious code can still be hide inside the correct content-type and execute when parsing).